OEL-3433: Move to ckeditor 5.

drishu opened the pull request:
## OEL-3433: Move to ckeditor 5

### Description

Update config and tests to ckeditor 5.

Most of the config is just a drush cde oe_showcase export, you can see that just the order of some elements changed in the yaml config.

I disabled ckeditor 4 so that we can remove it in the release after this is released.

drishu committed:
OEL-3433: Enable ckeditor5.

drishu committed:
OEL-3433: Use Ckeditor5 in text editors.

drishu committed:
OEL-3433: Adapt editor test to ckeditor5.

drishu committed:
OEL-3433: Disable ckeditor 4.

drishu committed:
OEL-3433: Adapt pressWysiwygButton() to ckeditor 5.

drishu committed:
OEL-3433: Adapt embed test to ckeditor 5.

drishu committed:
OEL-3433: Revert AV Portal assert.

drishu committed:
OEL-3433: Improve ck-content query selector.

donquixote committed:
OEL-3433: Don't replace existing embeds in WysiwygEmbedTest.

Changes:
- Insert a paragraph, to avoid replacing the previous embed.
- Wait until new embed appears in the editor area.

donquixote commented on config/install/views.view.media_entity_browsers.yml:
```diff
@@ -214,6 +214,7 @@ display:
             offset: false
             offset_label: Offset
           quantity: 9
+          pagination_heading_level: h4
```
Unrelated noise? Should we omit it from the PR or keep it in?
Personally, I like the following strategy:
- Have dedicated commits for config noise that has no functional consequence and that we are sure is reproducible.
  - Technically they would be "out of scope", but it is simply not practical to create dedicated issues.
  - This can even be split further into "reorder config keys" vs other noise.
  - One way to produce separate commits is to run config export _before_ doing the functional change. In this case, export the config and commit, before we enable ckeditor5.
- Do not include config noise that is not reproducible (e.g. depends on local env or external randomness)
- Do not include config noise that would cause functional changes or have some kind of risk. Instead, open separate issues.

donquixote commented on tests/src/ExistingSiteJavascript/WysiwygEmbedTest.php:
```diff
@@ -90,18 +89,18 @@ public function testMediaEmbed(): void {
-    $modal_button_pane = $assert_session->elementExists('css', 'div.ui-dialog-buttonpane');
+    // There are two embed dialogs, the second was the embed button.
+    $assert_session->waitForElementVisible('css', 'form.oe-oembed-entity-dialog-step--embed');
+    $modal_button_pane = $assert_session->elementExists('css', 'div.ui-dialog-buttonset');
```
Are some of these wait statements redundant?
(I don't think it is a blocker, just asking)

donquixote commented on tests/src/ExistingSiteJavascript/WysiwygEmbedTest.php:
```diff
@@ -90,18 +89,18 @@ public function testMediaEmbed(): void {
 
     // Check that the element has been inserted in the editor.
-    $this->getSession()->switchToIFrame('ckeditor');
     $assert_session->linkExists('Document title');
```
Interesting that this works without the wait above.
(ok for me)

donquixote committed:
OEL-3433: Update a module dependency to ckeditor5.

donquixote committed:
OEL-3433: Switch full_html to ckeditor5, even though it is disabled.

Keep most of the tags in allowed_html, except those that are redundant with enabled plugins.

donquixote committed:
OEL-3433: Disable editor.editor.full_html, which is a consequence of disabling the full_html text format.

The filter.format.full_html was disabled in the past.
But the disabling of the respective editor config was never exported.

donquixote committed:
OEL-3433: Disable ckeditor 4 _after_ editor config updates.

donquixote commented on config/install/editor.editor.full_html.yml:
This file is a bit special.

Before this PR:
The `filter.format.full_html` was already disabled, but the `editor.editor.full_html` was still enabled.
On re-enabling the text format, we see it was using ckeditor 4.

In a previous version of this PR:
Running updb would disable ckeditor module, which would remove the editor.editor.full_html (and any other editor config that used ckeditor 4).
Then the same updb would restore the new version of editor config for rich_text and simple_rich_text but not for full_html.
As a consequence, editor.editor.full_html was now missing in the database config, but still present in the config/install of oe_showcase, and in oe_showcase.info.yml.

In the new version of the PR:
I enabled the text format, switched the editor to ckeditor5, then disabled the text format again.

donquixote reviewed (commented):
I think all or most of this PR is good.

Also, for a regular website that moves to ckeditor5, they need to check how the editor will affect their existing content. But for oe_showcase we don't need to worry about this. Our default content does not have advanced markup like tables.

The next reviewer needs to check my latest commits.

drishu reviewed (commented):
Changed look good
